src/image_cache/image_loader.rs

```diff
@@ -150,6 +155,20 @@ pub fn load_gif(path: &Path, req_id: u32) -> Result<impl Iterator<Item = Result<
 	load_animation(req_id, decoder)
 }
 
+/// Parse, render and gather an SVG into a ImageBuffer<Rgba>
+pub fn load_svg(path: &std::path::Path) -> Result<image::RgbaImage> {
+	let opt = usvg::Options::default();
+	let rtree = usvg::Tree::from_file(path, &opt)?;
+	let (width, height) = (rtree.svg_node().size.width(), rtree.svg_node().size.height());
```

Comment:
May I suggest

```rust
let usvg::geom::Size { width, height } = rtree.svg_node().size;
```

Reply:
Changed. I believe that kind of pattern matching on let assignments is still WIP.

Reply:
[It's actually not](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=2606d7d379cc65a4b9c52b5bfe1a9970). The problem was that the fields of `Size` are not public. The reason for that is beyond me.